apps/console/src/middleware.ts

```diff
@@ -27,21 +27,22 @@ export default auth(async (req) => {
   const isTfaEnabled = session?.user.isTfaEnabled
 
   if (isLoggedIn) {
-    if (isTfaEnabled) {
+    if (isTfaEnabled && (path === '/tfa' || path === '/login')) {
+      return NextResponse.next()
+    } else if (isTfaEnabled) {
       return NextResponse.redirect(new URL('/tfa', req.url))
     }
 
     if (isPublicPage) {
       return NextResponse.redirect(new URL('/dashboard', req.url))
     }
-
     return NextResponse.next()
-  } else {
-    if (isPublicPage) {
-      return NextResponse.next()
-    }
-    return NextResponse.redirect(new URL('/login', req.url))
   }
+  // if not logged in
+  if (isPublicPage) {
+    return NextResponse.next()
+  }
+  return NextResponse.redirect(new URL('/login', req.url))
```

Comment:
Can make this a tenary operator to make it more readable (same for lines 36-39 but github won't let me suggest it because of a deleted line 🙃 
```suggestion
  // if not logged in
return isPublicPage ? NextResponse.next() : NextResponse.redirect(new URL('/login', req.url));
```